client/dfclient/src/websocket.cpp

```diff
@@ -90,79 +90,103 @@ bool WebSocketEmscripten::Send(std::string& data) {
 
 #else // __EMSCRIPTEN__
 
-#include <websocketpp/config/asio_no_tls_client.hpp>
-#include <websocketpp/client.hpp>
-
-typedef websocketpp::client<websocketpp::config::asio_client> WebSocketPPClient;
-
-class WebSocketPP
+class WebSocketBeast
 	: public WebSocket
 {
 public:
 	int Connect(std::string& address) override;
 	bool Send(std::string& data) override;
-	virtual ~WebSocketPP() {}
+	virtual ~WebSocketBeast() {}
 
-	WebSocketPPClient c;
-	websocketpp::connection_hdl myHdl;
 };
 
-typedef WebSocketPP WebSocketType;
+typedef WebSocketBeast WebSocketType;
 
-std::map<void*, WebSocketPP*> clients;
+#include <boost/beast/core.hpp>
+#include <boost/beast/websocket.hpp>
+#include <boost/asio/connect.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <cstdlib>
+#include <iostream>
+#include <string>
+#include <thread>
+
+namespace beast = boost::beast;         // from <boost/beast.hpp>
+namespace http = beast::http;           // from <boost/beast/http.hpp>
+namespace websocket = beast::websocket; // from <boost/beast/websocket.hpp>
+namespace net = boost::asio;            // from <boost/asio.hpp>
+using tcp = boost::asio::ip::tcp;       // from <boost/asio/ip/tcp.hpp>
+
+net::io_context ioc;
+websocket::stream<tcp::socket> ws{ioc};
+beast::flat_buffer buffer;
+
+void DoRead();
+
+void OnRead(beast::error_code ec, std::size_t bytes_transferred) {
+	if(ec == websocket::error::closed)
+		return;
+
+	if(ec) {
+		if (ec.value() == boost::system::errc::operation_canceled) {
+			// this websocket has disconnected
+			// todo: cleanup whatever is happening and return to main menu
+			return;
+		}
+		std::cout << "read failed " << ec << "\n";
+	}
 
-void WebSocketPPOnMessage(websocketpp::connection_hdl hdl, WebSocketPPClient::message_ptr msg) {
-	WebSocketPP* wspp = clients[hdl.lock().get()];
-	std::lock_guard<std::mutex> guard(wspp->mq_mutex);
-	wspp->messageQueue.push_back(msg->get_payload());
+	auto str = beast::buffers_to_string(buffer.data());
+	WebSocketBeast* wsb = (WebSocketBeast*) webSocketManager._ws.get();
+	std::lock_guard<std::mutex> guard(wsb->mq_mutex);
+	wsb->messageQueue.push_back(str);
+	buffer.consume(buffer.size());
+	DoRead();
 }
 
-void WebSocketPPOnOpen(websocketpp::connection_hdl hdl) {
-	WebSocketPP* wspp = clients[hdl.lock().get()];
-	std::lock_guard<std::mutex> guard(wspp->mq_mutex);
-	wspp->toBeOpened = true;
+void DoRead() {
+	ws.async_read(buffer, &OnRead);
 }
 
-int WebSocketPP::Connect(std::string& address) {
-	// Set logging to be pretty verbose (everything except message payloads)
-	c.set_access_channels(websocketpp::log::alevel::none);
-	c.set_error_channels(websocketpp::log::elevel::all);
+int WebSocketBeast::Connect(std::string& fullAddress) {
+	tcp::resolver resolver{ioc};
 
-	// Initialize ASIO
-	c.init_asio();
+	ws.binary(true);
 
-	// Register our message handler
-	c.set_message_handler(&WebSocketPPOnMessage);
-	c.set_open_handler(&WebSocketPPOnOpen);
+	auto address = fullAddress.substr(5, fullAddress.size());
```

Comment:
Why do we drop first 5 characters from the beginning?? I don't understand

Reply:
Changed 5 to where exactly it comes from